<html>
 <head>
  <meta charset="utf-8"/>
  <meta content="chrome=1" http-equiv="X-UA-Compatible"/>
  <title>
   Galaxy Docker Image by bgruening
  </title>
  <link href="css/landing_page.css" rel="stylesheet"/>
 </head>
 <body>
  <div class="wrapper">
   <header>
    <h1>
     Galaxy Docker Image
    </h1>
    <p>
     Docker Images tracking the stable Galaxy releases
    </p>
    <p class="bold">
     Table of content
    </p>
    <ul>
     <li>
      <a href="index.html">
       Global description
      </a>
     </li>
     <li>
      <a href="usage.html">
       Usage
      </a>
     </li>
     <li>
      <a href="auto-exec-tools-in-docker.html">
       Enable Galaxy to use BioContainers (Docker)
      </a>
     </li>
     <li>
      <a href="magic-environment-variables.html">
       Magic Environment variables
      </a>
     </li>
     <li>
      <a href="https-support.html">
       HTTPS Support
      </a>
     </li>
     <li>
      <a href="lite-mode.html">
       Lite Mode
      </a>
     </li>
     <li>
      <a href="extending-the-docker-image.html">
       Extending the Docker Image
      </a>
     </li>
     <li>
      <a href="integrating-non-tool-shed-tools-into-the-container.html">
       Integrating non-Tool Shed tools into the container
      </a>
     </li>
     <li>
      <a href="users-passwords.html">
       Users &amp; Passwords
      </a>
     </li>
     <li>
      <a href="development.html">
       Development
      </a>
     </li>
     <li>
      <a href="requirements.html">
       Requirements
      </a>
     </li>
     <li>
      <a href="support-bug-reports.html">
       Support &amp; Bug Reports
      </a>
     </li>
    </ul>
    <p class="view">
     <a href="https://github.com/bgruening/docker-galaxy">
      View the Project on GitHub
      <small>
       bgruening/docker-galaxy
      </small>
     </a>
    </p>
    <ul class="box">
     <li class="box">
      <a href="https://github.com/bgruening/docker-galaxy/zipball/master">
       Download
       <strong>
        ZIP File
       </strong>
      </a>
     </li>
     <li class="box">
      <a href="https://github.com/bgruening/docker-galaxy/tarball/master">
       Download
       <strong>
        TAR Ball
       </strong>
      </a>
     </li>
     <li class="box">
      <a href="https://github.com/bgruening/docker-galaxy">
       View On
       <strong>
        GitHub
       </strong>
      </a>
     </li>
    </ul>
   </header>
   <section>
    <h1>
     Extending the Docker Image
    </h1>
    <p>
     If the desired tools are already included in the Tool Shed, building your own personalised Galaxy docker Image (Galaxy flavour) can be done using the following steps:
    </p>
    <ol>
     <li>
      Create a file named
      <code>
       Dockerfile
      </code>
     </li>
     <li>
      Include
      <code>
       FROM quay.io/bgruening/galaxy
      </code>
      at the top of the file. This means that you use the Galaxy Docker Image as base Image and build your own extensions on top of it.
     </li>
     <li>
      Supply the list of desired tools in a file (
      <code>
       my_tool_list.yml
      </code>
      below). See
      <a href="https://github.com/galaxyproject/ansible-galaxy-tools/blob/master/files/tool_list.yaml.sample">
       this page
      </a>
      for the file format requirements.
     </li>
     <li>
      Execute
      <code>
       docker build -t my-docker-test .
      </code>
      4a. (if behind proxy). Add the ENV http_proxy and https_proxy variables as IPs (to avoid nameserver resolution problems) as in the example below.
     </li>
     <li>
      Run your container with
      <code>
       docker run -p 8080:80 my-docker-test
      </code>
     </li>
     <li>
      Open your web browser on
      <code>
       http://localhost:8080
      </code>
     </li>
    </ol>
    <p>
     For a working example, have a look at these  Dockerfiles.
    </p>
    <ul>
     <li>
      <a href="http://deeptools.github.io/">
       deepTools
      </a>
      <a href="https://github.com/bgruening/docker-recipes/blob/master/galaxy-deeptools/Dockerfile">
       Dockerfile
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/galaxytools/tree/master/chemicaltoolbox">
       ChemicalToolBox
      </a>
      <a href="https://github.com/bgruening/docker-recipes/blob/master/galaxy-chemicaltoolbox/Dockerfile">
       Dockerfile
      </a>
     </li>
    </ul>
    <pre><code># Galaxy - deepTools
#
# VERSION       0.2

FROM quay.io/bgruening/galaxy

MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com

ENV GALAXY_CONFIG_BRAND deepTools
# The following two lines are optional and can be given during runtime
# with the -e http_proxy='http://yourproxyIP:8080' parameter
ENV http_proxy 'http://yourproxyIP:8080'
ENV https_proxy 'http://yourproxyIP:8080'

WORKDIR /galaxy

RUN add-tool-shed --url 'http://testtoolshed.g2.bx.psu.edu/' --name 'Test Tool Shed'

# Install Visualisation
RUN install-biojs msa

# Adding the tool definitions to the container
ADD my_tool_list.yml $GALAXY_ROOT_DIR/my_tool_list.yml

# Install deepTools
RUN install-tools $GALAXY_ROOT_DIR/my_tool_list.yml

# Mark folders as imported from the host.
VOLUME ["/export/", "/data/", "/var/lib/docker"]

# Expose port 80 (webserver), 21 (FTP server)
EXPOSE :80
EXPOSE :21

# Autostart script that is invoked during container start
CMD ["/usr/bin/startup"]
</code></pre>
    <p>
     or the
     <a href="https://github.com/bgruening/galaxy-rna-workbench/blob/master/Dockerfile">
      RNA-workbench
     </a>
     .
The RNA-workbench has advanced examples about:
    </p>
    <ul>
     <li>
      <p>
       populating Galaxy data libraries
      </p>
      <pre><code class="language-bash">  setup-data-libraries -i $GALAXY_ROOT_DIR/library_data.yaml -g http://localhost:8080
      -u $GALAXY_DEFAULT_ADMIN_USER -p $GALAXY_DEFAULT_ADMIN_PASSWORD
</code></pre>
     </li>
    </ul>
    <p>
     The actual data is references in a YAML file similar this
     <a href="https://github.com/bgruening/galaxy-rna-workbench/blob/master/library_data.yaml">
      one
     </a>
     .
    </p>
    <ul>
     <li>
      <p>
       installing workflows
      </p>
      <pre><code class="language-bash">    workflow-install --workflow_path $GALAXY_HOME/workflows/ -g http://localhost:8080
        -u $GALAXY_DEFAULT_ADMIN_USER -p $GALAXY_DEFAULT_ADMIN_PASSWORD
</code></pre>
     </li>
    </ul>
    <p>
     Where all Galaxy workflows needs to be in one directory, here the
     <code>
      $GALAXY_HOME/workflows/
     </code>
     .
    </p>
    <ul>
     <li>
      <p>
       running Galaxy data-managers to create indices or download data
      </p>
      <pre><code class="language-bash">    run-data-managers -u $GALAXY_DEFAULT_ADMIN_USER -p $GALAXY_DEFAULT_ADMIN_PASSWORD -g http://localhost:8080
        --config data_manager_rna_seq.yaml
</code></pre>
     </li>
    </ul>
    <p>
     The data-managers can be configured and specified in a YAML file similar to this
     <a href="https://github.com/galaxyproject/training-material/blob/master/RNA-Seq/docker/data_manager_rna_seq.yaml">
      one
     </a>
     .
    </p>
    <p>
     If you host your flavor on GitHub consider to test our build with Travis-CI. This project will help you:
     <a href="https://github.com/bgruening/galaxy-flavor-testing">
      https://github.com/bgruening/galaxy-flavor-testing
     </a>
    </p>
    <h2>
     Test matrix
     <a name="Test-matrix">
     </a>
     <a href="#toc">
      [toc]
     </a>
    </h2>
    <p>
     The project includes local test scripts and CI workflows. Use the matrix below to decide what to run.
    </p>
    <table>
     <thead>
      <tr>
       <th>
        Area
       </th>
       <th>
        Script / Workflow
       </th>
       <th>
        Requires
       </th>
       <th>
        Notes
       </th>
      </tr>
     </thead>
     <tbody>
      <tr>
       <td>
        Image build
       </td>
       <td>
        <code>
         docker build -t galaxy:test galaxy/
        </code>
       </td>
       <td>
        Docker
       </td>
       <td>
        Baseline image build.
       </td>
      </tr>
      <tr>
       <td>
        Startup sanity
       </td>
       <td>
        <code>
         docker run --rm --privileged galaxy:test /usr/bin/startup2
        </code>
       </td>
       <td>
        Privileged
       </td>
       <td>
        Confirms services start and CVMFS messaging is sane.
       </td>
      </tr>
      <tr>
       <td>
        Bioblend
       </td>
       <td>
        <code>
         test/bioblend/test.sh
        </code>
       </td>
       <td>
        Running Galaxy container
       </td>
       <td>
        Uses a Bioblend test image against Galaxy.
       </td>
      </tr>
      <tr>
       <td>
        Slurm
       </td>
       <td>
        <code>
         test/slurm/test.sh
        </code>
       </td>
       <td>
        Docker, Slurm test image
       </td>
       <td>
        Uses external Slurm container; set
        <code>
         GALAXY_IMAGE=galaxy:test
        </code>
        if needed.
       </td>
      </tr>
      <tr>
       <td>
        SGE (Grid Engine)
       </td>
       <td>
        <code>
         test/gridengine/test.sh
        </code>
       </td>
       <td>
        Docker, SGE test image
       </td>
       <td>
        Uses ephemeris container to wait for Galaxy.
       </td>
      </tr>
      <tr>
       <td>
        CVMFS sidecar
       </td>
       <td>
        <code>
         test/cvmfs/test.sh
        </code>
       </td>
       <td>
        Privileged
       </td>
       <td>
        Builds and validates mount propagation from sidecar.
       </td>
      </tr>
      <tr>
       <td>
        FTP/SFTP
       </td>
       <td>
        <code>
         .github/workflows/single.sh
        </code>
       </td>
       <td>
        Docker, sshpass (CI)
       </td>
       <td>
        FTP and SFTP checks run in CI; local run skips SFTP if
        <code>
         sshpass
        </code>
        is missing.
       </td>
      </tr>
      <tr>
       <td>
        /export persistence
       </td>
       <td>
        <code>
         startup.sh
        </code>
        /
        <code>
         startup2.sh
        </code>
       </td>
       <td>
        <code>
         /export
        </code>
        volume
       </td>
       <td>
        Export and cache relocation happens during startup; exercised by CI runs.
       </td>
      </tr>
      <tr>
       <td>
        HTTPS/TLS
       </td>
       <td>
        <code>
         .github/workflows/single.sh
        </code>
       </td>
       <td>
        Docker
       </td>
       <td>
        Uses
        <code>
         curl
        </code>
        and
        <code>
         openssl s_client
        </code>
        against port 443.
       </td>
      </tr>
      <tr>
       <td>
        Tool install smoke
       </td>
       <td>
        <code>
         .github/workflows/single.sh
        </code>
       </td>
       <td>
        Docker
       </td>
       <td>
        Installs sample tools and verifies tool availability.
       </td>
      </tr>
      <tr>
       <td>
        Container resolvers
       </td>
       <td>
        <code>
         test/container_resolvers_conf.ci.yml
        </code>
       </td>
       <td>
        Galaxy container
       </td>
       <td>
        CI uses a minimal resolver config for toolbox resolution tests.
       </td>
      </tr>
      <tr>
       <td>
        Image analysis (optional)
       </td>
       <td>
        <code>
         .github/workflows/single.sh
        </code>
       </td>
       <td>
        <code>
         dive
        </code>
       </td>
       <td>
        Runs only when
        <code>
         dive
        </code>
        is installed.
       </td>
      </tr>
      <tr>
       <td>
        Single-container CI
       </td>
       <td>
        <code>
         .github/workflows/single_container.yml
        </code>
       </td>
       <td>
        CI
       </td>
       <td>
        Full container test (privileged).
       </td>
      </tr>
      <tr>
       <td>
        Multi-test CI
       </td>
       <td>
        <code>
         .github/workflows/single.sh
        </code>
       </td>
       <td>
        CI
       </td>
       <td>
        Builds image + runs SLURM, SGE, Bioblend; uses buildx cache.
       </td>
      </tr>
     </tbody>
    </table>
    <p>
     Notes:
    </p>
    <ul>
     <li>
      If
      <code>
       /tmp
      </code>
      is small in CI, set
      <code>
       TMPDIR=/var/tmp
      </code>
      for test scripts.
     </li>
     <li>
      CVMFS sidecar CI builds/pushes on tags; branch pushes run tests only when CVMFS paths change.
     </li>
    </ul>
    <h2>
     List of Galaxy flavours
     <a name="List-of-Galaxy-flavours">
     </a>
     <a href="#toc">
      [toc]
     </a>
    </h2>
    <ul>
     <li>
      <a href="https://github.com/statonlab/aurora-galaxy-tools">
       Aurora Galaxy
      </a>
     </li>
     <li>
      <a href="https://github.com/ValentinMarcon/docker-galaxy-sniplay">
       SNP analysis Workflows on Docker (sniplay)
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/docker-galaxy-blast">
       NCBI-Blast
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/docker-recipes/blob/master/galaxy-chemicaltoolbox">
       ChemicalToolBox
      </a>
     </li>
     <li>
      <a href="https://github.com/anhi/docker-scripts/tree/master/ballaxy">
       ballaxy
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/docker-recipes/blob/master/galaxy-deeptools">
       NGS-deepTools
      </a>
     </li>
     <li>
      <a href="https://github.com/gregvonkuster/docker-galaxy-ChIP-exo">
       Galaxy ChIP-exo
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/docker-galaxyp">
       Galaxy Proteomics
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/docker-galaxy-imaging">
       Imaging
      </a>
     </li>
     <li>
      <a href="https://github.com/gregvonkuster/docker-galaxy-csg">
       Constructive Solid Geometry
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/galaxy-metagenomics">
       Galaxy for metagenomics
      </a>
     </li>
     <li>
      <a href="https://github.com/lappsgrid-incubator/docker-galaxy-lappsgrid">
       Galaxy with the Language Application Grid tools
      </a>
     </li>
     <li>
      <a href="https://github.com/gianlucacorrado/galaxy-RNAcommender">
       RNAcommender
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/galaxy-open-molecule-generator">
       OpenMoleculeGenerator
      </a>
     </li>
     <li>
      <a href="https://github.com/workflow4metabolomics/w4m-docker">
       Workflow4Metabolomics
      </a>
     </li>
     <li>
      <a href="https://github.com/maxplanck-ie/docker-galaxy-hicexplorer">
       HiC-Explorer
      </a>
     </li>
     <li>
      <a href="https://github.com/phac-nml/snvphyl-galaxy">
       SNVPhyl
      </a>
     </li>
     <li>
      <a href="https://github.com/BackofenLab/docker-galaxy-graphclust">
       GraphClust
      </a>
     </li>
     <li>
      <a href="https://github.com/bgruening/galaxy-rna-workbench">
       RNA workbench
      </a>
     </li>
     <li>
      <a href="https://github.com/morinlab/tools-morinlab/tree/master/docker">
       Cancer Genomics Toolkit
      </a>
     </li>
     <li>
      <a href="http://cancerres.aacrjournals.org/content/77/21/e23">
       Clustered Heatmaps for Interactive Exploration of Molecular Profiling Data
      </a>
     </li>
    </ul>
   </section>
   <footer>
    <p>
     This project is maintained by
     <a href="https://github.com/bgruening">
      bgruening
     </a>
    </p>
    <p>
     <small>
      Hosted on GitHub Pages — Theme by
      <a href="https://github.com/orderedlist">
       orderedlist
      </a>
     </small>
    </p>
   </footer>
   <script src="js/landing_page.js">
   </script>
  </div>
 </body>
</html>
